src/main/java/com/hotels/beans/cache/CacheManager.java

```diff
@@ -72,17 +69,16 @@
      * @param <T> the class object type.
      * @return the cached object.
      */
+    @SuppressWarnings("unchecked")
     public <T> T getFromCache(final String cacheKey, final Class<? extends T> objectClass) {
-        notNull(cacheKey, "cacheKey cannot be null!");
```

Comment:
This is a non-trivial change for clients of the API (e.g. whether parameters can be null or not).  I am not sure whether this internal change should ripple outwards.  If the parameters are still required to be non-null, then perhaps adding https://github.com/uber/NullAway to the project would be useful.